src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -248,27 +223,23 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
             }
 
             if (error) {
-                reject(error);
+                return reject(error);
             }
 
-            if (paymentMethod.method === StripeElementType.CreditCard) {
-                resolve(stripePaymentMethod);
+            if (stripePaymentMethod && paymentMethod.method === StripeElementType.CreditCard) {
+                return resolve(stripePaymentMethod);
             }
 
-            resolve(paymentIntent);
+            if (paymentIntent) {
+                return resolve(paymentIntent);
+            }
         });
     }
 
-    private async _confirmVaultedPayment(clientSecret: string): Promise<PaymentIntent> {
-        return new Promise(async (resolve, reject) => {
-            const { error, paymentIntent } = await this._getStripeJs().handleCardAction(clientSecret);
+    private async _handleCustomerAction(clientSecret: string): Promise<PaymentIntent> {
+        const { paymentIntent } = await this._getStripeJs().handleCardAction(clientSecret);
```

Comment:
Is it not possible that this call answers an error as before?

Reply:
That's the PR's point. I removed on purpose the logic you mention because we need to reach the backend when an error happens in order to return the inventory.